### ISSUE TITLE
`IFS=` should be suggested for `while read -r line; do` to avoid omitting leading spaces

### ISSUE THREAD

**Issue by Cyberbeni** [koalaman/shellcheck#2619]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash

changes=()
while read -r line; do
	code="${line:0:2}"
	case "$code" in
		"??"|A?|?A|M?|?M)
			changes+=("${line:3}")
			;;
	esac
done < <(git status --porcelain=v1 --no-renames --ignore-submodules=all)
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Warning about omitted leading spaces and suggesting correcting to `while IFS= read -r line; do`
